src/main/java/org/mule/service/http/impl/service/client/GrizzlyHttpClient.java

```diff
@@ -331,8 +336,8 @@ private HttpResponse sendAndWait(HttpRequest request, HttpRequestOptions options
     CompletableFuture<HttpResponse> future = new CompletableFuture<>();
     try {
       AsyncHandler<Response> asyncHandler;
-      if (streamingEnabled) {
-        asyncHandler = new ResponseBodyDeferringAsyncHandler(future, responseBufferSize);
+      if (shouldStreamResponse(options)) {
+        asyncHandler = new ResponseBodyDeferringAsyncHandler(future, options.getResponseBufferSize().orElse(responseBufferSize));
```

Comment:
isn't the second param the same as calling `getPipeSize`?

Reply:
No, our handler tries to optimize the buffer size if not set, but the blocking one doesn't so in this case we don't want to set a default one.